src/routes/safe/components/Balances/Tokens/screens/AssetsList/AssetRow.tsx

```diff
@@ -3,43 +3,62 @@ import ListItemIcon from '@material-ui/core/ListItemIcon'
-import { withStyles } from '@material-ui/core/styles'
+import { makeStyles } from '@material-ui/core/styles'
 import React, { memo } from 'react'
 
 import { styles } from './style'
 
-import Img from 'src/components/layout/Img'
 import { ETH_ADDRESS } from 'src/logic/tokens/utils/tokenHelpers'
-import { setImageToPlaceholder } from 'src/routes/safe/components/Balances/utils'
 
+import { NFTAsset } from 'src/logic/collectibles/sources/OpenSea'
+
+import { TokenSymbol } from 'src/components/TokenSymbol'
 export const TOGGLE_ASSET_TEST_ID = 'toggle-asset-btn'
 
-// eslint-disable-next-line react/display-name
-const AssetRow = memo(({ classes, data, index, style }: any) => {
-  const { activeAssetsAddresses, assets, onSwitch } = data
-  const asset = assets.get(index)
-  const { address, image, name, symbol } = asset
-  const isActive = activeAssetsAddresses.has(asset.address)
-
-  return (
-    <div style={style}>
-      <ListItem classes={{ root: classes.tokenRoot }} className={classes.token}>
-        <ListItemIcon className={classes.tokenIcon}>
-          <Img alt={name} height={28} onError={setImageToPlaceholder} src={image} />
-        </ListItemIcon>
-        <ListItemText primary={symbol} secondary={name} />
-        {address !== ETH_ADDRESS && (
-          <ListItemSecondaryAction>
-            <Switch
-              checked={isActive}
-              inputProps={{ 'data-testid': `${symbol}_${TOGGLE_ASSET_TEST_ID}` } as any}
-              onChange={onSwitch(asset)}
-            />
-          </ListItemSecondaryAction>
-        )}
-      </ListItem>
-    </div>
-  )
-})
-
-export default withStyles(styles as any)(AssetRow)
+type Props = {
+  data: {
+    assets: Record<string, NFTAsset>
+    activeAssetsAddresses: Set<string>
+    onSwitch: (asset: NFTAsset) => () => void
+  }
+  style: unknown
+  index: number
+  isScrolling?: boolean
+}
+
+const useStyles = makeStyles(styles)
+
+const AssetRow = memo(
+  (props: Props): React.ReactElement => {
+    const { data, index, style } = props
+    const classes = useStyles()
+    const { activeAssetsAddresses, assets, onSwitch } = data
+    const asset = assets[index]
+    const { address, name, symbol } = asset
+    const isActive = activeAssetsAddresses.has(asset.address)
+
+    return (
+      <div style={style}>
+        <ListItem>
+          <ListItemIcon className={classes.tokenIcon}>
+            <TokenSymbol height={28} tokenAddress={asset.address} />
+          </ListItemIcon>
+          <ListItemText primary={symbol} secondary={name} />
+          {address !== ETH_ADDRESS && (
+            <ListItemSecondaryAction>
+              <Switch
+                checked={isActive}
+                inputProps={{ 'data-testid': `${symbol}_${TOGGLE_ASSET_TEST_ID}` } as unknown}
+                onChange={onSwitch(asset)}
+              />
+            </ListItemSecondaryAction>
+          )}
+        </ListItem>
+      </div>
+    )
+  },
+)
+
+AssetRow.displayName = 'AssetRow'
```

Comment:
out of ignorance, why is that?

Reply:
https://stackoverflow.com/questions/43356073/how-to-set-displayname-in-a-functional-component-react this thing :)

Reply:
oh, neat. Thanks!